include/kalmar_math.h

```diff
@@ -486,10 +486,10 @@ namespace Kalmar
         float fmod(float x, float y) { return fast_math::fmodf(x, y); }
 
         HCC_MATH_LIB_FN
-        float frexpf(float x, int *exp) { return __ocml_frexp_f32(x, exp); }
+        float frexpf(float x, int *exp) { return __ocml_frexp_f32(x, (__attribute__((address_space(5))) int*)exp); }
 
         HCC_MATH_LIB_FN
-        _Float16 frexp(_Float16 x, int *exp) { return __ocml_frexp_f16(x, exp); }
+        _Float16 frexp(_Float16 x, int *exp) { return __ocml_frexp_f16(x, (__attribute__((address_space(5))) int*) exp); }
```

Comment:
second definition in line 1160.

Reply:
.. different namespace.

Reply:
You can't cast a generic pointer to a private pointer and expect it to work unless the data being pointed at was allocated in private memory.  To handle all cases the implementation should look something like:  { int e; _Float16 ret = __ocml_frexp_f16(x, (__attribute__((address_space(5))) int*)&e); *exp = e; return ret; }
The compiler will usually clean up the extra load/store.

Reply:
Thanks Brian.  I think we'd need to do this for all of the functions with pass-by-reference args like this.

Reply:
I've pushed an updated commit that should resolve this.